### ISSUE TITLE
Ability to specify defaults by an annotation

### ISSUE THREAD

**Issue by bdrobinson** [sksamuel/avro4s#528]
I'd like to be able to specify a field as having a default value without having to use default arguments. Ie, currently I have to do:
```scala
case class MyRecord(
  some_field: Option[String] = None
)
```

whereas I'd like to be able to do something like:
```scala
case class MyRecord(
  @AvroDefault(None)
  some_field: Option[String]
)
```

The reason being that having the default value in the case class means it's very easy to forget to specify an argument in my application code when I'm instantiating the case class (which actually led to me nearly shipping a bug to production the other day). Would this be a difficult feature to add? I'd potentially be able to work on a PR – I work with @alexjg who's contributed to this codebase before.

Thanks very much!

**Comment by bdrobinson** [comment]
Not stale – I'd still be interested to see how hard this feature would be to add.

**Comment by janjaali** [comment]
Could we re-open this feature-request, having the same issue, or maybe a explanation why this is not do-able?

**Comment by janjaali** [comment]
Please don't stale, trying to find some time trying to handle this issue.

**Comment by sksamuel** [comment]
@janjaali if you want to add this, then look here: https://github.com/sksamuel/avro4s/blob/release/4.0.x/avro4s-core/src/main/scala/com/sksamuel/avro4s/Records.scala#L200-L205

You need to update that first case statement to look for an annotation on the field. You can add the lookup code in AnnotationExtractors.

**Comment by janjaali** [comment]
> @janjaali if you want to add this, then look here: https://github.com/sksamuel/avro4s/blob/release/4.0.x/avro4s-core/src/main/scala/com/sksamuel/avro4s/Records.scala#L200-L205
> 
> You need to update that first case statement to look for an annotation on the field. You can add the lookup code in AnnotationExtractors.

Thanks @sksamuel, I will try to come up with a PR in the upcoming holiday season. Wish you happy holidays and a good start into the new year ;)

**Comment by erikvanoosten** [comment]
Somewhat related feature request: #714.

**Comment by alexander-klimov** [comment]
I'm guessing this is not getting done any time soon? :(

**Comment by sksamuel** [comment]
If you pr I will release but I no longer actively work on scala projects.

**Comment by alexander-klimov** [comment]
> If you pr I will release but I no longer actively work on scala projects.

Nah, I'll just use [this solution](https://github.com/sksamuel/avro4s/issues/714#issuecomment-1092787180) by Erik
